polygeist/tools/cgeist/driver.cc

```diff
@@ -639,6 +639,10 @@ static int finalize(mlir::MLIRContext &context,
       module->walk([&](mlir::omp::ParallelOp) { LinkOMP = true; });
       mlir::PassManager pm3(&context);
       pm3.addPass(polygeist::createConvertToLLVMABIPass());
+      // Needed because SYCLMethodOps lowering might introduce redundant
+      // operations.
+      pm3.addPass(mlir::createCSEPass());
+      pm3.addPass(mlir::createCanonicalizerPass(canonicalizerConfig, {}, {}));
```

Comment:
Why cannot canonicalize at line 654 after `createConvertPolygeistToLLVMPass`?

Reply:
I guess there should be no reason not to do that.

Reply:
compile time?

Reply:
I'll change that on Monday.

Reply:
Thanks